src/lib/djerba/util/provenance_reader.py

```diff
@@ -90,11 +95,14 @@ def __init__(self, provenance_path, project, donor, samples,
         # read and parse file provenance
         self.logger.info("Reading provenance for project '%s' and donor '%s' " % (project, donor))
         self.root_sample_name = donor
-        if not samples.is_valid():
+        self.assay = assay
+        if not samples.is_valid(): 
             msg = "User-supplied sample names are not valid: {0}. ".format(samples)+\
                   "Requires {0} and {1} with optional {2}".format(self.wg_n, self.wg_t, self.wt_t)
             self.logger.error(msg)
             raise RuntimeError(msg)
+        if self.assay == self.TAR:
+            self._set_tar_ids(samples)
```

Comment:
This works, but doesn't make it clear how it is coupled with finding sample IDs for the not-TAR case.
Can we combine it into an if/else with this code on line 134?
```
            if assay != self.TAR:
                self._validate_and_set_sample_names(samples)
```